test/externalTests/sign.js

```diff
@@ -1,48 +1,104 @@
 const assert = require('assert')
 const Vec3 = require('vec3')
 
-module.exports = () => async (bot) => {
-  const mcData = require('minecraft-data')(bot.version)
-  const Item = require('prismarine-item')(bot.version)
-  const lowerBlock = bot.blockAt(bot.entity.position.offset(0, -1, 0))
+module.exports = (version) => {
+  const mcData = require('minecraft-data')(version)
+  const Item = require('prismarine-item')(version)
+
+  async function runTest (bot, testFunction) {
+    await testFunction(bot)
+  }
+
+  const tests = []
+
+  function addTest (name, f) {
+    tests[name] = bot => runTest(bot, f)
+  }
 
   let signItem = null
   for (const name in mcData.itemsByName) {
     if (name.includes('sign')) signItem = mcData.itemsByName[name]
   }
   assert.notStrictEqual(signItem, null)
 
-  const p = new Promise((resolve, reject) => {
-    bot._client.on('open_sign_entity', (packet) => {
-      const sign = bot.blockAt(new Vec3(packet.location))
-      bot.updateSign(sign, '1\n2\n3\n')
+  addTest('update sign', /** @param {import('mineflayer').Bot} bot */ async (bot) => {
+    const lowerBlock = bot.blockAt(bot.entity.position.offset(0, -1, 0))
+    await bot.lookAt(lowerBlock.position, true)
+    await bot.test.setInventorySlot(36, new Item(signItem.id, 1, 0))
+    const p = new Promise((resolve) => {
+      bot.on('signOpen', (packet) => {
+        const sign = bot.blockAt(new Vec3(packet.location))
+        bot.updateSign(sign, '1\n2\n3\n')
 
-      setTimeout(() => {
-        // Get updated sign
-        const sign = bot.blockAt(bot.entity.position)
+        setTimeout(() => {
+          // Get updated sign
+          const sign = bot.blockAt(bot.entity.position)
 
-        assert.strictEqual(sign.signText, '1\n2\n3\n')
+          assert.strictEqual(sign.signText, '1\n2\n3\n')
 
-        if (sign.blockEntity && sign.blockEntity.Text1) {
-          assert.strictEqual(sign.blockEntity.Text1.toString(), '1')
-          assert.strictEqual(sign.blockEntity.Text2.toString(), '2')
-          assert.strictEqual(sign.blockEntity.Text3.toString(), '3')
-          assert.strictEqual(sign.blockEntity.Text4.toString(), '')
-        }
+          if (sign.blockEntity && sign.blockEntity.Text1) {
+            assert.strictEqual(sign.blockEntity.Text1.toString(), '1')
```

Comment:
is there any way we can abstract over this? this is way too direct to have to use the sign's blockentity nbt.

Reply:
This test should be unnecessary anyway as we are check the sign.signText property